### PR TITLE
ExportContainer: do not panic

### PR DESCRIPTION
**Pull request by kolyshkin**
This is proposed replacement for #36563, fixing #36561 and a few more potential nil deref cases. Please see more detailed description in commit messages.

@vdemeester @tonistiigi @thaJeztah PTAL

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Review by lowenna — APPROVED** [review]
LGTM

File: container/archive.go [hunk 1]
```diff
@@ -6,13 +6,17 @@ import (
...
+	"github.com/pkg/errors"
 )
 
 // ResolvePath resolves the given path in the container to a resource on the
 // host. Returns a resolved path (absolute path to the resource on the host),
 // the absolute path to the resource relative to the container's rootfs, and
 // an error if the path points to outside the container's rootfs.
 func (container *Container) ResolvePath(path string) (resolvedPath, absPath string, err error) {
+	if container.BaseFS == nil {
+		return "", "", errors.New("ResolvePath: BaseFS of container " + container.ID + " is unexpectedly nil")
```
Comment (thaJeztah) [inline container/archive.go]: Wonder if we should prefix all of these with the function-name

**Review by tonistiigi — APPROVED** [review]

**Comment by thaJeztah** [comment]
Failure on z/s390x is a flaky test, and being addressed in https://github.com/moby/moby/pull/36551

**Comment by andrewhsu** [comment]
@kolyshkin if there isn't any test that covers the codebase affected by this PR, can we get unit tests for this in a separate PR?

File: container/archive.go (hunk 1 above)
Comment (kolyshkin) [reply container/archive.go]: The chance of such error is quite low (i.e. it should not happen with the current code base, as far as I was able to see), so normally a user won't even see it. If they do see it, though, it is kind of an internal error (as in "this software is buggy"), and having some context would be great to ease the debugging. Ideally, I would even add a stack trace here...

**Comment by kolyshkin** [comment]
> can we get unit tests for this in a separate PR?

OK, I will see if I can come up with a unit test, imitating a condition described in #36561

**Comment by kolyshkin** [comment]
> OK, I will see if I can come up with a unit test, imitating a condition described in #36561

Having said that, a unit test makes little sense here, it is much better to have an integration test. Here: https://github.com/moby/moby/pull/36606

**Pull request merged**